File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.empty()) return 0;
        set<int> set;
        for(auto &x : nums)
            set.emplace(x);
        auto it = set.begin();
        int longest = 1;
        int cur_seq = 1;
        int last = *set.begin();
        it++;
        for(;it != set.end(); it++) {
            int cur = *it;
            if (cur - last == 1)
                cur_seq++;
            else
                cur_seq = 1;
            last = cur;
            longest = max(longest, cur_seq); 
        }
        return longest;
    }
};

File: Data Structures & Algorithms/maximum-ascending-subarray-sum/submission-1.py
class Solution:
    def maxAscendingSum(self, nums: List[int]) -> int:
        curr_sum = nums[0]
        max_sum = 0
        i = 1
        if len(nums) == 1: return curr_sum
        
        while i < len(nums):
            max_sum = max(curr_sum, max_sum)
            if nums[i-1] < nums[i] :
                curr_sum += nums[i] 
            else:
                curr_sum = nums[i]
            print(curr_sum)
            max_sum = max(curr_sum, max_sum)
            i += 1

        return max_sum